dev/com.ibm.ws.microprofile.faulttolerance/test/src/com/ibm/ws/microprofile/faulttolerance/test/BulkheadTest.java

```diff
@@ -238,40 +238,37 @@ public void testAsyncBulkhead() throws InterruptedException, ExecutionException,
     }
 
     @Test
-    public void testAsyncBulkheadQueueFull() throws InterruptedException, ExecutionException, TimeoutException {
+    public void testAsyncBulkheadDefaultsAndException() throws InterruptedException, ExecutionException, TimeoutException {
         BulkheadPolicy bulkhead = FaultToleranceProvider.newBulkheadPolicy();
-        bulkhead.setMaxThreads(2);
-        bulkhead.setQueueSize(2);
```

Comment:
Can you set a different size for queue and thread, then change the size of the future?

Reply:
But then it will no longer test the default values, which were previously not being set properly?  Maybe I'm not following what you mean.

Reply:
Perhaps the test should be duplicated rather than just modified ... one to test the defaults and one to test the overrides

Reply:
The test before this one, testAsyncBulkhead(), already does overrides.  The test I modified looked like it was only testing that the bulkhead exception was thrown when the queue was full.  This is retained in the modified test, while also testing the defaults.  With that said, I'm happy to just add another test, but it will be exactly like the one I added since the exception must be triggered to validate the correct default for the queue.  I'm also happy to hand this over to you guys since you're the SME regarding this code.

Reply:
In order for that to work, you would have to be sure that you were able to submit the 21st execution before any of the previous 20 have completed. The approx execution time was 2000ms. That was safe enough when you only had to submit 4 but I'm not so sure about 20. Also, you have removed the 100ms sleep from line 260 ... that was there to give the async framework time to pull work from the queue and begin execution. With a queue size of 10, you'll only be able to submit 20 executions if the framework has had time to pull the first 10 off the queue.